### PR TITLE
LibWeb+LibXML: Make SVG file loaded directly

### PR DESCRIPTION
**Pull request by zhiyuang**
The error happens when svg file starts with:
```
<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
```

The parser did not handle this properly before.

Before:
Screenshot 2023-07-12 at 21 47 11

After:
Screenshot 2023-07-12 at 21 40 27

### CODE REVIEW

File: Userland/Libraries/LibWeb/XML/XMLDocumentBuilder.cpp [hunk 1]
```diff
@@ -22,6 +22,9 @@ ErrorOr<DeprecatedString> resolve_xml_resource(XML::SystemID const&, Optional<XM
         return Error::from_string_literal("Refusing to load disallowed external entity");
 
     auto public_literal = public_id->public_literal;
+    if (public_literal.is_one_of("-//W3C//DTD SVG 1.1//EN")) {
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/XML/XMLDocumentBuilder.cpp:25]: No need to use `is_one_of` with a single argument. Just compare with `==` :)

Comment (alimpfard) [inline Userland/Libraries/LibWeb/XML/XMLDocumentBuilder.cpp:25]: This bit is supposed to be in accordance with the HTML spec, we probably shouldn't even be trying to build an html document for an svg+xml file

File: Userland/Libraries/LibXML/Parser/Parser.cpp [hunk 2]
```diff
@@ -409,6 +409,7 @@ ErrorOr<void, ParseError> Parser::parse_standalone_document_decl()
     TRY(expect("standalone"sv));
     auto accept = accept_rule();
 
+    TRY(parse_eq());
```
Comment (alimpfard) [inline Userland/Libraries/LibXML/Parser/Parser.cpp:412]: Please move this to a separate commit.

**Review by alimpfard — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibWeb/XML/XMLDocumentBuilder.cpp (hunk 1 above)
Comment (alimpfard) [reply Userland/Libraries/LibWeb/XML/XMLDocumentBuilder.cpp:25]: For reference, the list of acceptable identifiers is given in <https://html.spec.whatwg.org/multipage/xhtml.html#parsing-xhtml-documents> (though an SVG file is _not_ an XHTML file, so it shouldn't even get to this line)

Comment (zhiyuang) [reply Userland/Libraries/LibWeb/XML/XMLDocumentBuilder.cpp:25]: thanks for the detailed explanation 👍

File: Userland/Libraries/LibXML/Parser/Parser.cpp (hunk 2 above)
Comment (zhiyuang) [reply Userland/Libraries/LibXML/Parser/Parser.cpp:412]: Ok, I will close this PR and start a new one with this change.
